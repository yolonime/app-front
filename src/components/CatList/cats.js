const cats = [
  {
    id: "1r1",
    url: "https://cdn2.thecatapi.com/images/1r1.jpg",
    width: 500,
    height: 335,
  },
  {
    id: "6gp",
    url: "https://cdn2.thecatapi.com/images/6gp.jpg",
    width: 500,
    height: 333,
  },
  {
    id: "a4a",
    url: "https://cdn2.thecatapi.com/images/a4a.jpg",
    width: 525,
    height: 700,
  },
  {
    id: "bp3",
    url: "https://cdn2.thecatapi.com/images/bp3.jpg",
    width: 500,
    height: 381,
  },
  {
    id: "bqa",
    url: "https://cdn2.thecatapi.com/images/bqa.jpg",
    width: 960,
    height: 638,
  },
  {
    id: "drt",
    url: "https://cdn2.thecatapi.com/images/drt.jpg",
    width: 690,
    height: 494,
  },
  {
    id: "dvt",
    url: "https://cdn2.thecatapi.com/images/dvt.jpg",
    width: 700,
    height: 518,
  },
  {
    id: "MTUxMDkwNw",
    url: "https://cdn2.thecatapi.com/images/MTUxMDkwNw.jpg",
    width: 500,
    height: 333,
  },
  {
    id: "MTY0NTY3Mw",
    url: "https://cdn2.thecatapi.com/images/MTY0NTY3Mw.jpg",
    width: 640,
    height: 425,
  },
  {
    id: "MTk0MTcyOQ",
    url: "https://cdn2.thecatapi.com/images/MTk0MTcyOQ.jpg",
    width: 690,
    height: 941,
  },
];

export default cats;
