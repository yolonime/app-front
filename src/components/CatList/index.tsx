import { FC, useEffect, useState } from "react";
import { get } from "../../utils/api";

type Cat = {
  id: string;
  url: string;
  width: number;
  height: number;
};

const CatList: FC = () => {
  const [cats, setCats] = useState<Cat[]>([]);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const data = await get(
          "https://api.thecatapi.com/v1/images/search?limit=10"
        );

        setCats(data);
      } catch (e) {
        console.log(e);
      }
    };

    fetchData();
  }, []);

  return (
    <div className="cat-list">
      <span>Some Random Cats</span>
      {cats.map((cat) => {
        return (
          <div key={cat.id}>
            <img width="200" src={cat.url} height="200" alt="" />
          </div>
        );
      })}
    </div>
  );
};

export default CatList;
