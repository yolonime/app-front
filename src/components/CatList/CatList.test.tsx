import { render, waitFor, screen } from "@testing-library/react";
import mockData from "./cats";
import CatList from "./";

jest.mock("../../utils/api", () => {
  return {
    get: () => Promise.resolve(mockData),
  };
});

describe("Cat List component", () => {
  it("should display title", async () => {
    render(<CatList />);

    expect(screen.getByText("Some Random Cats")).toMatchSnapshot();
  });

  it("should fetch cats when button is clicked", async () => {
    const { baseElement } = render(<CatList />);
    await waitFor(() => {});
    expect(baseElement).toMatchSnapshot();
  });
});
