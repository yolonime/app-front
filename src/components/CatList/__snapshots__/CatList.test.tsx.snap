// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cat List component should display title 1`] = `
<span>
  Some Random Cats
</span>
`;

exports[`Cat List component should fetch cats when button is clicked 1`] = `
<body>
  <div>
    <div
      class="cat-list"
    >
      <span>
        Some Random Cats
      </span>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/1r1.jpg"
          width="200"
        />
      </div>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/6gp.jpg"
          width="200"
        />
      </div>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/a4a.jpg"
          width="200"
        />
      </div>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/bp3.jpg"
          width="200"
        />
      </div>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/bqa.jpg"
          width="200"
        />
      </div>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/drt.jpg"
          width="200"
        />
      </div>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/dvt.jpg"
          width="200"
        />
      </div>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/MTUxMDkwNw.jpg"
          width="200"
        />
      </div>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/MTY0NTY3Mw.jpg"
          width="200"
        />
      </div>
      <div>
        <img
          alt=""
          height="200"
          src="https://cdn2.thecatapi.com/images/MTk0MTcyOQ.jpg"
          width="200"
        />
      </div>
    </div>
  </div>
</body>
`;
